packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx

```diff
@@ -96,8 +75,29 @@ export function LocationDetailView({
     onSearchQueryChange,
     onSearchClear,
     onToggleSearchSubfolders,
+    hasExhaustedSearch,
   } = useLocationDetailView({ onNavigate: onNavigateProp, onExit });
 
+  let messageControlContent: MessageProps | undefined;
+
+  if (hasError) {
+    messageControlContent = getListItemsResultMessage({
+      items: pageItems,
+      hasError,
+      message,
+    });
+  } else if (hasExhaustedSearch) {
+    messageControlContent = getListItemsResultMessage({
+      items: pageItems,
+      hasExhaustedSearch,
+      message,
+    });
+  } else {
+    messageControlContent = getListItemsResultMessage({
+      items: pageItems,
+    });
+  }
```

Comment:
We should not be providing params to `getListItemsResults` conditionally. `getListItemsResults` should always be provided with all parameters and decide when to return a message, same comment for `LocationsView`. Can be updated in a follow up.